src/nspawn/nspawn.c

```diff
@@ -3486,6 +3487,16 @@ static int inner_child(
                 n_env++;
         }
 
+        if (arg_start_mode != START_BOOT && !strv_isempty(arg_parameters)) {
```

Comment:
why the check for arg_parameters hre?

Reply:
Just launching a shell session (without passing arguments) already derives the locale from /etc/locale.conf so I figure we don't want to change that.

Reply:
that's a fedora hack in their /etc/profile, no?

I'd not rely on that. it's a weird hack I think, and unlikely to be available on all distros...

Reply:
Yeah, let's drop the special check.